crates/flo/src/poison.rs

```diff
@@ -0,0 +1,66 @@
+//! Support functionality for generically poisonable types.
+
+use crate::types::{
+    ArrayType,
+    Block,
+    Diagnostic,
+    Location,
+    MatchArm,
+    PoisonType,
+    Statement,
+    StructType,
+    Variable,
+};
+
+/// Trait for FLO values that can be considered poisoned.
+pub trait Poisonable {
+    /// Returns a poison value for a T with the given `PoisonType`.
+    fn get_poison_value(typ: PoisonType) -> Self;
+
+    /// Returns whether or not the given value is poisoned.
+    fn is_poisoned(value: &Self) -> bool;
```

Comment:
A use case hasn't occured, and the `.poison` field values are publicly checkable. 

Unless you think it's likely we'll want that specifically, I'd say we make that easy addition if/when required? That way we get an API tailored to our actual needs later, rather than an API built for what we think we might need now.